src/generate.py

```diff
@@ -6,7 +6,32 @@
 from datetime import datetime
 from models import RetNetModel, TransformerModel
 from transformers import PreTrainedTokenizerFast
-from utils import Struct, generate_text
+from utils import Struct, generate_text, generate_text_from_tokens
+
+def generate_text_length_n(config: Struct, n: int, input_tokens: torch.Tensor):
```

Comment:
I never saw input_tokens become anything but a python list of random number representing tokens from our vocabulary; is there a reason it's been type-hinted as a `torch.Tensor`?